bpfd-operator/controllers/bpfd-agent/tc-program.go

```diff
@@ -186,41 +188,50 @@ func (r *TcProgramReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 			return ctrl.Result{Requeue: true, RequeueAfter: retryDurationAgent}, nil
 		}
 
-		retry, err := reconcileProgram(ctx, r, r.currentTcProgram, &r.currentTcProgram.Spec.BpfProgramCommon, r.ourNode, existingPrograms)
+		result, err := reconcileProgram(ctx, r, r.currentTcProgram, &r.currentTcProgram.Spec.BpfProgramCommon, r.ourNode, existingPrograms)
 		if err != nil {
-			r.Logger.Error(err, "Reconciling TcProgram Failed", "TcProgramName", r.currentTcProgram.Name, "Retrying", retry)
-			return ctrl.Result{Requeue: retry, RequeueAfter: retryDurationAgent}, nil
+			r.Logger.Error(err, "Reconciling TcProgram Failed", "TcProgramName", r.currentTcProgram.Name, "ReconcileResult", ReconcileResultToString(result))
+		}
+		switch result {
```

Comment:
This exact logic is duplicated for every bpfd-agent controller right? If so is it easy to internalize into a function in common.go

Reply:
I know.  It bothered me too, but was trying not to do too much surgery.  However, I'll take another look.

Reply:
I'm not sure if you're just referring to the switch statement or the whole function.  While the switch statement is duplicated, it's controlling the flow of the loop, so I don't think I can just offload that part.  A macro would work, but Go doesn't have them.  I did summarize the comments because I was pretty much duplicating the comments on the result constants.  Actually, almost all the code in this whole file is the same for all the program types except for the types.  Do you have any ideas?